cloud/scope/common_test.go

```diff
@@ -0,0 +1,180 @@
+package scope
+
+import (
+	"context"
+	"errors"
+	"testing"
+
+	"github.com/linode/linodego"
+	"github.com/stretchr/testify/assert"
+	"go.uber.org/mock/gomock"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/linode/cluster-api-provider-linode/mock"
+)
+
+// Test_createLinodeClient tests the createLinodeClient function. Checks if the client does not error out.
+func TestCreateLinodeClient(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		name   string
+		apiKey string
+		want   *linodego.Client
+	}{
+		{
+			"Valid API Key",
+			"test-key",
+			createLinodeClient("test-key"),
+		},
+		{
+			"Empty API Key",
+			"",
+			createLinodeClient(""),
+		},
+	}
+
+	for _, tt := range tests {
+		testCase := tt
+		t.Run(testCase.name, func(t *testing.T) {
+			t.Parallel()
+
+			if got := createLinodeClient(testCase.apiKey); got != nil {
+				assert.EqualExportedValues(t, testCase.want, got, "Checking is the objects are equal")
+			}
+		})
+	}
+}
+
+// Test_getCredentialDataFromRef tests the getCredentialDataFromRef function.
+func TestGetCredentialDataFromRef(t *testing.T) {
+	t.Parallel()
+
+	type args struct {
+		providedCredentialsRef corev1.SecretReference
+		expectedCredentialsRef corev1.SecretReference
+		funcBehavior           func(ctx context.Context, key types.NamespacedName, obj *corev1.Secret, opts ...client.GetOption) error
+	}
+
+	tests := []struct {
+		name          string
+		args          args
+		expectedByte  []byte
+		expectedError string
+	}{
+		{
+			name: "Check is the function works correctly",
```

Comment:
The test description is a bit generic. Can you describe what's specific to this test case? i.e. "gets an apiToken from the Secret in a different namespace"

Reply:
Updated!